lib/Pool.js

```diff
@@ -264,16 +272,33 @@ Pool.prototype._removeWorker = function(worker) {
  *                                  after finishing all tasks currently in
  *                                  progress. If true, the workers will be
  *                                  terminated immediately.
+ * @param {number} [timeout]        If provided and non-zero, worker termination promise will be rejected
+ *                                  after timeout if worker process has not been terminated.
+ * @return {Promise.<void, Error>}
  */
-// TODO: rename clear to terminate
-Pool.prototype.clear = function (force) {
-  this.workers.forEach(function (worker) {
-    // TODO: implement callbacks when a worker is actually terminated, only then clear the worker from our array
-    //       else we get zombie child processes :)
-    worker.terminate(force);
+Pool.prototype.terminate = function (force, timeout) {
+  var f = function (worker) {
+    this._removeWorkerFromList(worker);
+  };
+  const removeWorker = f.bind(this);
```

Comment:
I would prefer to use `var` only and not `const`, since we don't have a transpiler in place which will make the library work on older browsers too (IE 10).

Reply:
OK. I have been writing in Typescript mostly with tslint, which is very opinionated, and that has trained me to use const without thinking. var is fine.